### PR TITLE
Configuration templates (part 2)

### PR DESCRIPTION
**Pull request by sbesson**
Follow-up of https://github.com/openmicroscopy/openmicroscopy/pull/4093, this PR:

- implements the substitutions of `@Ice.Default.Host@` for the configuration templates
- suppresses the help of `admin ports` 
- moves the Web templates to a subdirectory for clarity. The `web config` subcommand and the unit tests are updated accordingly

The CI servers will test the deployment of the servers on a non-default `Ice.Default.Host`. This PR should also be tested locally with `Ice.Default.Host` unset. Additionally, the Web configuration generation should be tested to make sure nothing was broken by the templates migration.

### CODE REVIEW

File: components/tools/OmeroPy/src/omero/plugins/admin.py [hunk 1]
```diff
@@ -320,24 +320,7 @@ def __init__(this, name, help, wait=False):
...
-
-  # Set ports to registry:14061, tcp:14063, ssl:14064, web:14080
-  %(prog)s --prefix=1
-  # Set ports back to defaults: 4061, 4063, 4064, 4080
-  %(prog)s --prefix=1 --revert
-  # Set ports to: 4444, 5555, 6666, 7777
-  %(prog)s --registry=4444 --tcp=5555 --ssl=6666 --webserver=7777
-
-""").parser
+        ports = Action("ports", SUPPRESS).parser
```
Comment (aleksandra-tarkowska) [inline components/tools/OmeroPy/src/omero/plugins/admin.py]: ``bin/omero admin ports`` are deprecated. Why just not add warning to the help as well.

Comment (sbesson) [inline components/tools/OmeroPy/src/omero/plugins/admin.py]: This came from [this discussion](https://github.com/openmicroscopy/openmicroscopy/pull/4093#commitcomment-13534609). The main point is that this command is now a no-op and adding `(deprecated)` to the help does not convey exactly that messafe.

**Comment by aleksandra-tarkowska** [comment]
Looks good to me

**Comment by sbesson** [comment]
Last commit should fix the failing unit test in https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-matlab/57/

**Comment by mtbc** [comment]
OMERO-DEV-merge-matlab has been passing lately.

**Comment by jburel** [comment]
Such properly could potentially be used in insight container.

**Comment by jburel** [comment]
Some tests have hard coded, localhost. do you plan to modify them in this PR and take advantage of the variable?

**Comment by sbesson** [comment]
@jburel: which tests do you have in mind specifically? The `Ice.Default.Host` variable is a server-side property. For client/test consumption, generally I would expect to read the hostname directly from `ice.config`.

**Comment by jburel** [comment]
Sorry I should have made it clearer, I was thinking of test that did not read directly from ice.config.

**Comment by jburel** [comment]
All good. tests are passing. I need to check if not set. I will check with @aleksandra-tarkowska

**Comment by jburel** [comment]
Question:
in ```etc/templates/ice.config```, you set the ```omero.host``` to ```@Ice.Default.Host@````
Do we want to do the same for ```omero.fs.host``` in ```etc/templates/grid/templates.xml ````?

**Comment by sbesson** [comment]
It would make sense to me. Is `omero.fs.host` expected to be modified by configuration property @ximenesuk?

**Comment by sbesson** [comment]
As pointed out by @aleksandra-tarkowska, OMERO.py and OMERO.web are currently relying on the presence of a configuration file under `etc/ice.config`. The last commits restore a default file with values `localhost/4064` which:
- is consumed by the build system for packaging OMERO.matlab and OMERO.py
- removes the burden of maintaining the substitution logic in the build system
- re-allows the following workflow  `./build.py build-dev && dist/bin/omero web start`

The other functionalities of the CLI admin plugin should be unaffected, i.e. the `etc/ice.config` file should be rewritten when calling `bin/omero admin rewrite` or `bin/omero admin start`

**Comment by ximenesuk** [comment]
@sbesson I don't fully understand the question. Do you mean modified by `omero config`?

**Comment by sbesson** [comment]
@ximenesuk: yes, is `omero.fs.host` an internal variable only or could it be adjusted via `omero config`. We do not expose it in the documentation.

**Comment by ximenesuk** [comment]
@sbesson Yes, it's not exposed on the DropBox docs - whether that makes it internal or not I'm not sure. Thoughts @joshmoore ?

**Comment by joshmoore** [comment]
The most likely use of that would be if someone wanted to run `DropBox` elsewhere:

 * `wget OMERO.server.zip`
 * ...edit `etc/grid` ...
 * `bin/omero config set omero.fs.host foo`
 * `bin/omero node dropbox`

**Comment by sbesson** [comment]
Unless there is a clear blocker on Dropbox and assuming the last commits make sense,I would think we can get this PR as it is and include Dropbox as part of 5.1.0-rc1 testing.

**Comment by joshmoore** [comment]
WFM

**Comment by ximenesuk** [comment]
...and me.

**Pull request merged**